app/js/disease-associations-controller.js

```diff
@@ -68,34 +68,13 @@ angular.module('cttvControllers')
 
         // facets changed?
         if (!_.isEqual(new_state[facetsId], old_state[facetsId]) || !new_state[facetsId]) {
-            getFacets(new_state[facetsId]);
+            $scope.getFacets(new_state[facetsId]);
         }
 
     };
 
-    var initFilterByFile =function(){
-        $scope.fileName = "";
+    $scope.targetArray = [];
```

Comment:
Then can communicate from directive to directive by sharing the targetArray. Something like:
controller:
$scope.targetArray = [];  // Initialisation (although not strictly neccessary

template:
<filter-by-target-list targets="targetArray"></filter-by-target-list-targets>
<associations-table targets="targetArray"></associations-table>

You need 2-way data binding on your directive, so when you change the array inside, it gets propagated outside and the table directive watch the change